src/AwkwardArray.jl

```diff
@@ -1,5 +1,109 @@
 module AwkwardArray
 
-# Write your package code here.
+abstract type Content end
+
+function Base.iterate(x::Content)
+    start = firstindex(x)
+    stop = lastindex(x)
+    if stop >= start
+        x[start], start + 1
+    else
+        nothing
+    end
+end
+
+function Base.iterate(x::Content, state)
+    stop = lastindex(x)
+    if stop >= state
+        x[state], state + 1
+    else
+        nothing
+    end
+end
+
+### PrimitiveArray #######################################################
+
+struct PrimitiveArray{T} <: Content
+    data::AbstractArray{T,1}
+end
+
+function is_valid(x::PrimitiveArray)
+    true
+end
+
+function Base.length(x::PrimitiveArray)
+    Base.length(x.data)
+end
+
+function Base.firstindex(x::PrimitiveArray)
+    Base.firstindex(x.data)
+end
+
+function Base.lastindex(x::PrimitiveArray)
+    Base.lastindex(x.data)
+end
+
+function Base.getindex(x::PrimitiveArray, i::Int)
+    x.data[i]
+end
+
+function Base.getindex(x::PrimitiveArray, r::UnitRange{Int})
+    PrimitiveArray(x.data[(r.start):(r.stop)])
```

Comment:
```suggestion
    PrimitiveArray(x.data[r])
```

since `x.data` is also <: `AbstractArray`, it should just work

Reply:
Fixed it; thanks!